Firestore/core/src/firebase/firestore/remote/remote_event.h

```diff
@@ -309,7 +311,7 @@ class RemoteEvent {
 class WatchChangeAggregator {
  public:
   explicit WatchChangeAggregator(
-      id<FSTTargetMetadataProvider> target_metadata_provider)
+      TargetMetadataProvider* target_metadata_provider)
```

Comment:
Similar to my other comment, consider making nullability explicit here. (i.e. docstring and notnull assertion if relevant.)

(Similarly here, you've technically changed the behaviour, since obj-c is quite happy to allow nils without segfaulting. But again, you're probably now more in line with the other ports, so this is entirely fine.)

Update: there's a bunch of other functions that just pipe this value through. Adding docstrings to those functions too might have some value in a public api, but personally, I think that would just be noise for internal apis. You could even omit the docstring here... the assertion itself would be a stronger form of documentation. :)